service/history/api/create_workflow_util.go

```diff
@@ -167,6 +167,7 @@ func NewWorkflowWithSignal(
 	return newMutableState, nil
 }
 
+// NOTE: must implement CreateLeaseFunc.
 func NewWorkflowLeaseAndContext(
```

Comment:
```suggestion
func NewWorkflowLease(
```
?
Yes, it creates new context internally but returns only lease.

Reply:
![](https://user-images.githubusercontent.com/2232524/75076254-832f8f00-54b4-11ea-81b9-1a6648a486a5.png)